File: backend/src/main/java/ajou/db/gpt/constant/Category.java
package ajou.db.gpt.constant;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;

@Schema(description = "<p>질문 카테고리. " +
        "<p>카테고리는 다음 항목들로 구성되어 있다." +
        "<ul>" +
        "<li><code>SCIENCE</code> - 과학/기술" +
        "<li><code>DEV</code> - 개발/코딩" +
        "<li><code>HISTORY</code> - 역사" +
        "<li><code>ART</code> - 예술/문학" +
        "<li><code>SOCIAL_SCIENCE</code> - 사회과학" +
        "<li><code>SPORTS</code> - 스포츠" +
        "<li><code>CULTURE</code> - 여행/문화" +
        "<li><code>HEALTH</code> - 건강" +
        "<li><code>NATURE</code> - 자연/환경" +
        "<li><code>ETC</code> - 기타" +
        "</ul>",
        example = "SCIENCE")
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public enum Category {

    SCIENCE("과학/기술"),
    DEV("개발/코딩"),
    HISTORY("역사"),
    ART("예술/문학"),
    SOCIAL_SCIENCE("사회과학"),
    SPORTS("스포츠"),
    CULTURE("여행/문화"),
    HEALTH("건강"),
    NATURE("자연/환경"),
    ETC("기타");

    private final String description;
}

File: frontend/src/components/util/category.ts
export const categoryList = [
  { key: "전체", val: "" },
  { key: "과학/기술", val: "SCIENCE" },
  { key: "개발/코딩", val: "DEV" },
  { key: "역사", val: "HISTORY" },
  { key: "예술/문학", val: "ART" },
  { key: "사회과학", val: "SOCIAL_SCIENCE" },
  { key: "스포츠", val: "SPORTS" },
  { key: "여행/문화", val: "CULTURE" },
  { key: "건강", val: "HEALTH" },
  { key: "자연/환경", val: "NATURE" },
  { key: "기타", val: "ETC" },
];

export const categoryObj: { [key: string]: string } = {
  전체: "",
  "과학/기술": "SCIENCE",
  "개발/코딩": "DEV",
  역사: "HISTORY",
  "예술/문학": "ART",
  사회과학: "SOCIAL_SCIENCE",
  스포츠: "SPORTS",
  "여행/문화": "CULTURE",
  건강: "HEALTH",
  "자연/환경": "NATURE",
  기타: "ETC",
};

export const SortedVal: { [key: string]: string } = {
  최신순: "LATEST",
  "오래된 순": "OLDEST",
};
